lms/djangoapps/course_api/serializers.py

```diff
@@ -0,0 +1,125 @@
+"""
+Course API Serializers.  Representing course catalog data
+"""
+
+import urllib
+
+from django.core.urlresolvers import reverse
+from django.template import defaultfilters
+
+from rest_framework import serializers
+
+from xmodule.course_module import DEFAULT_START_DATE
+from lms.djangoapps.courseware.courses import course_image_url, get_course_about_section
+
+
+class _MediaSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Nested serializer to represent a media object.
+    """
+    def __init__(self, name=u'', description=u'', type_=u'', uri_parser=lambda x: x, *args, **kwargs):
+        super(_MediaSerializer, self).__init__(*args, **kwargs)
+
+        self.name_value = name
+        self.description_value = description
+        self.type_value = type_
+        self.uri_parser = uri_parser
+
+    uri = serializers.SerializerMethodField(source='*')
+    name = serializers.SerializerMethodField()
+    description = serializers.SerializerMethodField()
+    type = serializers.SerializerMethodField()  # pylint: disable=redefined-builtin
+
+    def get_name(self, _):
+        """
+        Get the representation for the media resource's name
+        """
+        return self.name_value
+
+    def get_description(self, _):
+        """
+        Get the representation for the media resource's description
+        """
+        return self.description_value
+
+    def get_type(self, _):
+        """
+        Get the representation for the media resource's type
+        """
+        return self.type_value
+
+    def get_uri(self, course):
+        """
+        Get the representation for the media resource's URI
+        """
+        return self.uri_parser(course)
+
+
+class _CourseApiMediaCollectionSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Nested serializer to represent a collection of media objects
+    """
+    course_image = _MediaSerializer(
+        source='*',
+        name=u'Course Image',
+        description=u'',
+        type_=u'logo',
+        uri_parser=course_image_url
+    )
+
+
+class CourseSerializer(serializers.Serializer):  # pylint: disable=abstract-method
+    """
+    Serializer for Course objects
+    """
+
+    course_id = serializers.CharField(source='id', read_only=True)  # pylint: disable=invalid-name
+    name = serializers.CharField(source='display_name_with_default')
+    number = serializers.CharField(source='display_number_with_default')
+    org = serializers.CharField(source='display_org_with_default')
+    short_description = serializers.SerializerMethodField()
```

Comment:
@jcdyer If other parties agree, my preference is to name this "description".  If we ever have any other types of descriptions, we can annotate those accordingly.